experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -257,3 +257,97 @@ function isomorphism(::Type{PermGroup}, W::WeylGroup; set_properties::Bool=true)
+from `nothing`, and otherwise `one(genimgs[1])` is returned.
+
+See also: [`map_word(::Union{FPGroupElem, SubFPGroupElem}, ::Vector)`](@ref),
+[`map_word(::Union{PcGroupElem, SubPcGroupElem}, ::Vector)`](@ref).
+Note that `map_word(::WeylGroupElem)` does not provide a `genimgs_inv` parameter
+because the generators of a Weyl group are always self-inverse.
+
+# Examples
+```jldoctest
+julia> W = weyl_group(:B, 3); imgs = [2, 3, 5];
+
+julia> map_word(one(W), imgs)
+1
+
+julia> map_word(W([1]), imgs)
+2
+
+julia> map_word(W([1]), imgs, init=7)
+14
+
+julia> map_word(W([1,2,1]), imgs)
+12
+
+julia> map_word(W([2,1,2]), imgs) # W([2,1,2]) == W([1,2,1])
+12
+
+julia> map_word(W([3, 2, 1, 3, 2, 3]), imgs)
+2250
+```
+"""
+function map_word(w::WeylGroupElem, genimgs::Vector; init=nothing)
+  @req length(genimgs) == number_of_generators(parent(w)) "Length of vector of images does not equal rank of Weyl group"
+  return Oscar.map_word([Int(i) for i in word(w)], genimgs, init=init)
+end
+
+@doc raw"""
+    parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem) -> WeylGroup, Map{WeylGroup, WeylGroup}
+
+Returns two objects: The subgroup `U` of `W` generated by `[inv(w)*u*w for u in gens(W)[vec]]`, whose generators are set to be this vector and which is itself a `WeylGroup`, and the embedding of `U` into `W`. The elements of `vec` must be pairwise distinct integers in `1:number_of_generators(W)` and `vec` must be non-empty.
+"""
+function parabolic_subgroup(W::WeylGroup, vec::Vector{<:Integer}, w::WeylGroupElem=one(W))
+  @req length(unique(vec)) == length(vec) "Elements of vector are not pairwise distinct"
+  cm = cartan_matrix(root_system(W))[vec, vec]
+  para = weyl_group(cm)
+  genimgs = map(x -> inv(w) * x * w, gens(W)[vec])
+  emb = function(u::WeylGroupElem)
+    return map_word(u, genimgs)
+  end
+  return para, MapFromFunc(para, W, emb)
+end
+
+@doc raw"""
+    parabolic_subgroup_with_projection(W::WeylGroup, vec::Vector{<:Integer}) -> WeylGroup, Map{WeylGroup, WeylGroup}, Map{WeylGroup, WeylGroup}
+
+Returns a triple `(U, emb, proj)` that describes a factor of `W`, that is, a product of irreducible factors.
+Here the parabolic subgroup `U` and its embedding into `W` are as returned by [`parabolic_subgroup`](@ref)`(W, vec)`
+and `proj` is the projection map from `W` onto `U`.
+If `U` is not a factor of `W`, an error occurs.
+"""
+function parabolic_subgroup_with_projection(W::WeylGroup, vec::Vector{<:Integer}; check_factor::Bool=true)
+  if check_factor
+    # Check that every generator in gens(W)[vec] commutes with every other generator.
+    # In other words, vec describes a union of irreducible components of the Coxeter diagram.
+    cm = cartan_matrix(root_system(W))
+    for i in vec
+      for j in setdiff(1:number_of_generators(W), vec)
+        @req coxeter_matrix_entry_from_cartan_matrix(cm, i, j) == 2 "Input vector does not describe a factor of the Weyl group, so there is no projection homomorphism"
```

Comment:
```suggestion
        @req is_zero_entry(cm, i, j) "Input vector does not describe a factor of the Weyl group, so there is no projection homomorphism"
```
Is this mathematically the same? (I am not sure atm.) If yes, this would be a bit more performant

Reply:
I'm pretty sure it's mathematically the same (`cm(i, j) == 0` iff the simple roots with indices `i` and `j` are orthogonal iff $s_i$ commutes with $s_j$), so I made this change. Also wrapped the line here.